RecoBTag/FeatureTools/src/deep_helpers.cc

```diff
@@ -162,11 +162,17 @@ namespace btagbtvdeep {
 
     assert(min <= pad_value && pad_value <= max);
 
-    for (unsigned i = 0; i < input.size(); ++i) {
-      datavec.push_back(std::clamp((catch_infs(input[i], replace_inf_value) - center) * norm_factor, min, max));
+    if (input.size() <= target_length) {
+      for (unsigned i = 0; i < input.size(); ++i) {
```

Comment:
rather than duplicating the entire for loop, just make the condition `i < std::min(input.size(),target_length)`

Reply:
Now that you mention it, that should have been obvious.  Thanks!  I push the change